iotilesensorgraph/iotile/sg/parser/statements/copy_statement.py

```diff
@@ -95,7 +95,13 @@ def execute(self, sensor_graph, scope_stack):
             op = 'copy_count_a'
 
         if self.explicit_input:
-            sensor_graph.add_node(u"({} always && {} {}) => {} using {}".format(self.explicit_input, trigger_stream, trigger_cond, self.output, op))
+            #If root node is an input, create an intermediate node with an unbuffered node
+            if 'input' in str(self.explicit_input):
```

Comment:
Again, this should check the type of the selector.

Reply:
What's the difference between a DataStream and DataStreamSelector? Sometimes, self.explicit_input is just a DataStream so it doesn't have a selector variable. They both do have a .stream_type variable which I can use to check if it's a DataStream.InputType.

Reply:
DataStream can only name a single stream like `input 15`.  A selector is a criteria used to select streams so it could select a single stream in which case it's the same `input 15' but it could select a whole class of streams like `all inputs`.